src/engine/readaheadmanager.cpp

```diff
@@ -101,47 +104,26 @@ int ReadAheadManager::getNextSamples(double dRate, CSAMPLE* buffer,
 
             // do crossfade from the current buffer into the new loop beginning
             if (samples_read != 0) { // avoid division by zero
-                double mix_amount = 0.0;
-                double mix_inc = 2.0 / static_cast<double>(samples_read);
-                for (int i = 0; i < samples_read; i += 2) {
-                    base_buffer[i] = base_buffer[i] * (1.0 - mix_amount) + m_pCrossFadeBuffer[i] * mix_amount;
-                    base_buffer[i+1] = base_buffer[i+1] * (1.0 - mix_amount) + m_pCrossFadeBuffer[i+1] * mix_amount;
-                    mix_amount += mix_inc;
-                }
+                SampleUtil::linearCrossfadeBuffers(base_buffer, base_buffer, m_pCrossFadeBuffer, samples_read);
             }
         }
     }
 
     // Reverse the samples in-place
     if (in_reverse) {
-        // TODO(rryan) pull this into MixxxUtil or something
-        CSAMPLE temp1, temp2;
-        for (int j = 0; j < samples_read/2; j += 2) {
-            const int endpos = samples_read-1-j-1;
-            temp1 = base_buffer[j];
-            temp2 = base_buffer[j+1];
-            base_buffer[j] = base_buffer[endpos];
-            base_buffer[j+1] = base_buffer[endpos+1];
-            base_buffer[endpos] = temp1;
-            base_buffer[endpos+1] = temp2;
-        }
+        SampleUtil::reverse(base_buffer, samples_read);
     }
 
     //qDebug() << "read" << m_iCurrentPosition << samples_read;
     return samples_read;
 }
 
-void ReadAheadManager::addEngineControl(EngineControl* pControl) {
-    if (pControl == NULL) {
-        return;
-    }
-    m_sEngineControls.append(pControl);
+void ReadAheadManager::addLoopingControl(LoopingControl* pLoopingControl) {
+    m_pLoopingControl = pLoopingControl;
```

Comment:
Ha yes, LoopinControl ins mandantory so I will move the setter and the assertion to the constructor.